source/core/Ky.ts

```diff
@@ -145,10 +149,9 @@ export class Ky {
 		};
 
 		const isRetriableMethod = ky._options.retry.methods.includes(ky.request.method.toLowerCase());
-		const result = isRetriableMethod ? ky._retry(fn) : fn();
+		const result = (isRetriableMethod ? ky._retry(fn) : fn()) as ResponsePromise;
 
-		for (const [type, mimeType] of Object.entries(responseTypes)) {
-			// @ts-expect-error not sure how to properly type this!
+		for (const [type, mimeType] of Object.entries(responseTypes) as ObjectEntries<typeof responseTypes>) {
```

Comment:
the problem here is that `Object.entries` has incorrect `ReturnType`

So what I do:

1. I created `ObjectEntries` helper to infer it (however, it's possible to redefine it globally)
2. `result` as `ResponsePromise` now has all the methods for `type` so we can call it without issues

Reply:
Sorry to give you merge conflicts. I added `entry as [keyof typeof responseTypes, string]` in 
https://github.com/sindresorhus/ky/pull/355/commits/3e75d74d7b0eb6385d19d6ff1d080eb0f0e7834a#diff-1d28e040cea335f5a22aa9c3f678f5d00b353e3ce966dc0978a91fe56f240b04R151-L152

Reply:
Good, I will update the PR on the weekend, should be fine 🔥